### PR TITLE
BREAKING(http): remove Go-style address

### LINKED ISSUE #1642: http: Remove Go-style address/port strings
Early on in the lifetime of Deno, we used strings like `":8000"` to declare a server listening on localhost at port 8000. However long ago Deno's internal APIs no longer parse these strings. See for example [`Deno.ListenOptions`](https://doc.deno.land/builtin/stable#Deno.ListenOptions). 

However in std/http/server.ts and elsewhere we still accept strings of this form or use them internally. [`_parseAddrFromStr`](https://github.com/denoland/deno_std/blob/1767fa0990acd58dea139aab65a5deadbdf3eeff/http/server.ts#L43-L87) should be removed, and probably the necessary changes to do that would be enough.

### PR DESCRIPTION
**Pull request by jonasgrunert**
This hopefully fixes #1642. Currently this PR simply removes the functionality and replaces it through the Deno.ListenOptions aka {port: number, hostnam: string}.
An open question would be wether the option should be removed or only deprecated. Additionally the legacy HTTP server utilizes the same options aka currently the go style address. Right now this is also migrated towards the ListenerOptions. Should the options for the legacy server also be adapted or remain in the go style?

### CODE REVIEW

**Comment by kt3k** [comment]
@jonasgrunert 
Thank you for your contribution! Mostly looks good to me. Please format the code with `./_tools/format.js`.

Also the CI checks show type errors in example docs. Can you address those errors?

> An open question would be wether the option should be removed or only deprecated.

I think removal is ok as these APIs are relatively young.

> Additionally the legacy HTTP server utilizes the same options aka currently the go style address. Right now this is also migrated towards the ListenerOptions. Should the options for the legacy server also be adapted or remain in the go style?

We're going to remove the entire legacy server API very soon. ref: https://github.com/denoland/deno_std/pull/1648 It'd be helpful if you keep those file as is

**Comment by jonasgrunert** [comment]
I hope I was able to adress the concerns and fix the errrors.

**Comment by jonasgrunert** [comment]
I finally was able to get my local setup to properly execute documentation tests. I hope I now resolved errors. Sorry for the inconvenience.

**Comment by bartlomieju** [comment]
@jonasgrunert no problem. It looks like the CI failure is unrelated to your changes and caused by incomplete release process of Deno v1.16.4. I'll fix it.

**Review by kt3k — APPROVED** [review]
@jonasgrunert LGTM. Nice work!

**Review by bartlomieju — APPROVED** [review]
LGTM too

File: http/server.ts [hunk 1]
```diff
@@ -622,8 +543,12 @@ export async function serve(
   handler: Handler,
   options: ServeInit = {},
 ): Promise<void> {
-  const addr = options.addr ?? ":8000";
-  const server = new Server({ addr, handler, onError: options.onError });
+  const server = new Server({
+    port: options.port ?? 8000,
+    hostname: options.hostname ?? "0.0.0.0",
+    handler,
+    onError: options.onError,
```
Comment (cmorten) [inline http/server.ts:550]: An aside to this PR (highlighted due to formatting changes) - curious that have the `onError` option for `serve()` but not `serveTls()`

File: http/server.ts [hunk 2]
```diff
@@ -714,16 +639,16 @@ export async function serveTls(
  * });
  * ```
  *
- * @param addr The address to listen on.
+ * @param addr The Deno.ListenOptions to specify the hostname and port.
```
Comment (cmorten) [inline http/server.ts:642]:
```suggestion
 * @param config The Deno.ListenOptions to specify the hostname and port.
```

File: http/server.ts [hunk 3]
```diff
@@ -693,19 +618,19 @@ export async function serveTls(
...
  * ```ts
  * import { listenAndServe } from "https://deno.land/std@$STD_VERSION/http/server.ts";
  *
- * const addr = ":4505";
+ * const port = 4505;
  *
  * console.log("server listening on http://localhost:4505");
  *
- * await listenAndServe(addr, (request) => {
+ * await listenAndServe({port}, (request) => {
```
Comment (cmorten) [inline http/server.ts:633]:
```suggestion
 * await listenAndServe({ port }, (request) => {
```

(formatting docs same as `deno fmt` would for non-docs code)

File: http/server.ts [hunk 4]
```diff
@@ -693,19 +618,19 @@ export async function serveTls(
  * incoming connections, and handles requests on these connections with the
  * given handler.
  *
- * If the port is omitted from the address, `:80` is used.
+ * If the port is omitted from the ListenOptions, 80 is used.
  *
- * If the host is omitted from the address, the non-routable meta-address
+ * If the host is omitted from the ListenOptions, the non-routable meta-address
```
Comment (cmorten) [inline http/server.ts:623]: Consider either `Deno.ListenOptions` (aka the full type) or `config` (the name of the argument in the function signature)

File: http/server.ts [hunk 5]
```diff
@@ -739,21 +664,21 @@ export async function listenAndServe(
...
  *
- * const addr = ":4505";
+ * const port = 4505;
  * const certFile = "/path/to/certFile.crt";
  * const keyFile = "/path/to/keyFile.key";
  *
  * console.log("server listening on http://localhost:4505");
  *
- * await listenAndServeTls(addr, certFile, keyFile, (request) => {
+ * await listenAndServeTls({port}, certFile, keyFile, (request) => {
```
Comment (cmorten) [inline http/server.ts:681]:
```suggestion
 * await listenAndServeTls({ port }, certFile, keyFile, (request) => {
```

File: http/server.ts [hunk 6]
```diff
@@ -762,20 +687,20 @@ export async function listenAndServe(
  * });
  * ```
  *
- * @param addr The address to listen on.
+ * @param addr The Deno.ListenOptions to specify the hostname and port.
```
Comment (cmorten) [inline http/server.ts:690]:
```suggestion
 * @param config The Deno.ListenOptions to specify the hostname and port.
```

File: http/server.ts (hunk 1 above)
Comment (kt3k) [reply http/server.ts:550]: Right. That was a known issue and addressed at #1679

**Pull request merged**